bn-apps/cordapp-updates-distribution/corda-updates-app/src/main/kotlin/net/corda/cordaupdates/app/bno/ReportCordappVersionFlowResponder.kt

```diff
@@ -0,0 +1,59 @@
+package net.corda.cordaupdates.app.bno
+
+import co.paralleluniverse.fibers.Suspendable
+import net.corda.cordaupdates.app.member.CordappVersionInfo
+import net.corda.cordaupdates.app.member.ReportCordappVersionFlow
+import net.corda.core.flows.FlowLogic
+import net.corda.core.flows.FlowSession
+import net.corda.core.flows.InitiatedBy
+import net.corda.core.flows.StartableByRPC
+import net.corda.core.identity.CordaX500Name
+import net.corda.core.utilities.ProgressTracker
+import net.corda.core.utilities.unwrap
+
+@InitiatedBy(ReportCordappVersionFlow::class)
+class ReportCordappVersionFlowResponder(private val session : FlowSession) : FlowLogic<Unit>() {
+    @Suspendable
+    override fun call() {
+        val cordappVersionInfo = session.receive<CordappVersionInfo>().unwrap { it }
+        val dbService = serviceHub.cordaService(DatabaseService::class.java)
+        dbService.updateCordappVersionInfo(session.counterparty, cordappVersionInfo)
+    }
+}
+
+@StartableByRPC
+class GetCordappVersionsFlow : FlowLogic<Map<String, List<CordappVersionInfo>>>() {
+    companion object {
+        val QUERYING_FROM_DATABASE = ProgressTracker.Step("Querying data from the database")
```

Comment:
Probably not necessary now Stefano fixed the shell so you can have an empty progress tracker (again, if you're targeting C4).

Reply:
That's great. Added TODOs to remove them once v4 is released